common-lib/src/main/scala/com/gu/mediaservice/lib/logging/GridLogging.scala

```diff
@@ -0,0 +1,36 @@
+package com.gu.mediaservice.lib.logging
+import com.gu.mediaservice.lib.auth.ApiAccessor
+import com.typesafe.scalalogging.{Logger, StrictLogging}
+import net.logstash.logback.marker.Markers
+
+import scala.collection.JavaConverters._
+
+trait GridLogging extends StrictLogging {
+  case class ImageId(id: String)
+
+  implicit class LoggerWithHelpers(logger: Logger) {
```

Comment:
Just out of interest, and if you've got time – I'm unsure about this implicit class. Is it effectively an override for an implicit provided in `StrictLogging` (where we provide some sugar for common ops)?

Reply:
The implicit class is scala's mechanism for support extension methods. This is essentially adding the methods in the body of the class to the entity in the single parameter constructor. This is useful if you want to add a method onto a class that you can't edit the code of (i.e. in a third party library). It's all resolved at compile time so there isn't a runtime overhead. This is the third overloaded use of `implicit` in scala and doesn't relate to the other uses of implicit.

Reply:
Ah, gotcha – thanks for the explanation. Yes, that's not something I would have guessed, because as you say ...
> This is the third overloaded use of implicit in scala and doesn't relate to the other uses of implicit.

😅

Reply:
Think they are changing this in dotty, so hopefully it will become more obvious. 🤞